lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -157,6 +157,11 @@ sub _setup_websocket_connection ($self, $websocket_url = undef) {
 
                 my $error = $tx->error;
                 my $error_message = "Unable to upgrade to ws connection via $websocket_url";
```

Comment:
I would just alter the URL here:
```suggestion
                my $error_message = 'Unable to upgrade to ws connection via ' . $tx->req->url;
```
The original URL is still logged in the "Establishing …" line so there's nothing lost in my opinion.